internal/rest/avi_obj_vs.go

```diff
@@ -632,88 +642,83 @@ func (rest *RestOperations) AviVsVipBuild(vsvip_meta *nodes.AviVSVIPNode, cache_
 		// handling static IP updates, this would throw an error
 		// for advl4 the error is propagated to the gateway status
 		if vsvip_meta.IPAddress != "" {
-			auto_alloc := true
-			var vips []*avimodels.Vip
-			vips = append(vips, &avimodels.Vip{
+			vip := &avimodels.Vip{
 				VipID:          &vipId,
-				AutoAllocateIP: &auto_alloc,
+				AutoAllocateIP: &autoAllocate,
 				IPAddress: &avimodels.IPAddr{
 					Type: &ipType,
 					Addr: &vsvip_meta.IPAddress,
 				},
-			})
-			vsvip.Vip = vips
+			}
+			if networkName != "" {
+				vip.IPAMNetworkSubnet = &avimodels.IPNetworkSubnet{
+					NetworkRef: &networkRef,
+				}
+			}
+			vsvip.Vip = []*avimodels.Vip{vip}
 		}
 
-		path = "/api/vsvip/" + cache_obj.Uuid
-		rest_op = utils.RestOp{Path: path, Method: utils.RestPut, Obj: vsvip,
-			Tenant: vsvip_meta.Tenant, Model: "VsVip", Version: utils.CtrlVersion}
+		rest_op = utils.RestOp{
+			Path:    "/api/vsvip/" + cache_obj.Uuid,
+			Method:  utils.RestPut,
+			Obj:     vsvip,
+			Tenant:  vsvip_meta.Tenant,
+			Model:   "VsVip",
+			Version: utils.CtrlVersion,
+		}
 	} else {
-		auto_alloc := true
-		var vips []*avimodels.Vip
-
-		// all vsvip models would have auto_alloc set to true even in case of static IP programming
-		vip := avimodels.Vip{AutoAllocateIP: &auto_alloc}
-		networkRef := lib.GetNetworkName()
-		if lib.IsPublicCloud() && lib.GetNetworkName() != "" {
+		// all vsvip models would have AutoAllocateIP set to true even in case of static IP programming
+		vip := avimodels.Vip{
+			VipID:          &vipId,
+			AutoAllocateIP: &autoAllocate,
+		}
 
+		// setting IPAMNetworkSubnet.Subnet value in case subnetCIDR is provided
+		if lib.GetSubnetPrefix() == "" || subnetAddress == "" {
+			utils.AviLog.Warnf("Incomplete values provided for subnetIP, will not use IPAMNetworkSubnet in vsvip")
+		} else if lib.IsPublicCloud() {
```

Comment:
this can further be simplified with this condition `lib.GetCloudType() == lib.CLOUD_GCP`

Reply:
done